src/components/ContextMenu/index.tsx

```diff
@@ -0,0 +1,570 @@
+  useDialogControl as useContextMenuControl,
+} from '#/components/Dialog'
+
+const {Provider: PortalProvider, Outlet, Portal} = createPortalGroup()
+
+const SPRING: WithSpringConfig = {
+  mass: isIOS ? 1.25 : 0.75,
+  damping: 150,
+  stiffness: 1000,
+  restDisplacementThreshold: 0.01,
+}
+
+/**
+ * Needs placing near the top of the provider stack, but BELOW the theme provider.
+ */
+export function Provider({children}: {children: React.ReactNode}) {
+  return (
+    <PortalProvider>
+      {children}
+      <Outlet />
+    </PortalProvider>
+  )
+}
+
+export function Root({children}: {children: React.ReactNode}) {
+  const [measurement, setMeasurement] = useState<Measurement | null>(null)
+  const animationSV = useSharedValue(0)
+  const translationSV = useSharedValue(0)
+  const isFocused = useIsFocused()
+
+  const clearMeasurement = useCallback(() => setMeasurement(null), [])
+
+  const context = useMemo<ContextType>(
+    () => ({
+      isOpen: !!measurement && isFocused,
+      measurement,
+      animationSV,
+      translationSV,
+      open: (evt: Measurement) => {
+        setMeasurement(evt)
+        animationSV.set(withSpring(1, SPRING))
+      },
+      close: () => {
+        animationSV.set(
+          withSpring(0, SPRING, finished => {
+            if (finished) {
+              translationSV.set(0)
+              runOnJS(clearMeasurement)()
+            }
+          }),
+        )
+      },
+    }),
+    [
+      measurement,
+      setMeasurement,
+      isFocused,
+      animationSV,
+      translationSV,
+      clearMeasurement,
+    ],
+  )
+
+  useEffect(() => {
+    if (isAndroid && context.isOpen) {
+      const listener = BackHandler.addEventListener('hardwareBackPress', () => {
+        context.close()
+        return true
+      })
+
+      return () => listener.remove()
+    }
+  }, [context])
+
+  return <Context.Provider value={context}>{children}</Context.Provider>
+}
+
+export function Trigger({children, label, contentLabel, style}: TriggerProps) {
+  const context = useContextMenuContext()
+  const playHaptic = useHaptics()
+  const {top: topInset} = useSafeAreaInsets()
+  const ref = useRef<View>(null)
+  const isFocused = useIsFocused()
+  const [image, setImage] = useState<string | null>(null)
+  const [pendingMeasurement, setPendingMeasurement] =
+    useState<Measurement | null>(null)
+
+  const open = useNonReactiveCallback(async () => {
+    playHaptic()
+    Keyboard.dismiss()
+    const [measurement, capture] = await Promise.all([
+      new Promise<Measurement>(resolve => {
+        ref.current?.measureInWindow((x, y, width, height) =>
+          resolve({
+            x,
+            y:
+              y +
+              platform({
+                default: 0,
+                android: topInset, // not included in measurement
+              }),
+            width,
+            height,
+          }),
+        )
+      }),
+      captureRef(ref, {result: 'data-uri'}),
```

Comment:
Similar question, any chance of anything in here throwing? Could we bail on the trigger clone view and just show the menu in that case?